### PR TITLE
feat:[RP-21] - Button with variants

### PR DESCRIPTION
**Pull request by dipSyxx**
# Title
Button with variants

## Ticket Link
https://rozetka-home-project.atlassian.net/browse/RP-21

## Changes
-Added demo variants
-Added to the button href,type, disabled, onClick
-Added some styles to the global styles for the button
## Info
-Custom button with clear understanding wich variants we have
## Screenshots
FINAL RESULT:

Screenshot_1

### CODE REVIEW

**Comment by MaksPapirovnyk** [comment]
Screenshots of UI, to understand the final result

File: src/elements/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import clsx from 'clsx'
+import { createElement, ElementType, FC, ReactNode } from 'react'
+
+type ButtonProps = {
+  children: ReactNode
+  component: ElementType
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:6]: `component: ElementType` so the button can be any HTML element or maybe `<a>`(if we have a link) and `<button>`

File: src/elements/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  variant?: 'green' | 'linkButton' | 'outlined'
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:7]: `variant?: 'green' | 'linkButton' | 'outlined'`
separate buttons by usage, for example, the most popular is `primary`,  then maybe `secondary`
if you want to have a link, just use the `href` props and if passed a link so you can automatically pick up the `href` variant

File: src/elements/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  classVariant: string
+  href?: string
+  onClick?: () => void
+  type?: 'submit'
+  disabled?: 'isValid' | boolean
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:12]: `disabled?: boolean`

File: src/elements/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import { createElement, ElementType, FC, ReactNode } from 'react'
+
+type ButtonProps = {
+  children: ReactNode
+  component: ElementType
+  variant?: 'green' | 'linkButton' | 'outlined'
+  classVariant: string
+  href?: string
+  onClick?: () => void
+  type?: 'submit'
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:11]: ` type?: 'submit'` lets leave `type="button"` hardcoded and remove this props

File: src/elements/AuthModalSocials/AuthModalSocials.tsx [hunk 5]
```diff
@@ -6,11 +6,10 @@ export const AuthModalSocials = () => {
       <div className="auth_modal_socials_buttons">
         <div className="auth_modal_socials_label">Увійти як користувач</div>
         <button className="button button_socials">
-          {/* TODO: collapse empty tag */}
-          <i className="facebookicon fa-brands fa-facebook-f"></i> Facebook
+          <i className="fa-brands fa-facebook-f" /> Facebook
```
Comment (MaksPapirovnyk) [inline src/elements/AuthModalSocials/AuthModalSocials.tsx:9]: looks like you made changes for another ticket, don't touch parts of a code if it's already in another merge request or in progress, you may cause conflicts

File: src/styles/globalButtonsStyle.sass [hunk 6]
```diff
@@ -0,0 +1,268 @@
+//!variant
+
+//! classVariant ===============================|
+a
```
Comment (MaksPapirovnyk) [inline src/styles/globalButtonsStyle.sass:4]: the same about global styles and modifying code of anther PR

File: src/elements/Button/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import clsx from 'clsx'
+import { createElement, ElementType, FC, ReactNode } from 'react'
+
+import styles from './Button.module.sass'
+
+type ButtonProps = {
+  children: ReactNode
+  component: ElementType
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'href' | 'href_'
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:9]: `| 'href' | 'href_'` why? :) what does it mean?

File: src/elements/Button/Button.tsx (hunk 1 above)
Comment (MaksPapirovnyk) [reply src/elements/Button/Button.tsx:8]: Why you resolved this?

File: src/elements/Button/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  classVariant: string
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.tsx:10]: `classVariant ` what does it mean? is it a specific variant classname? or it's just a custom classname for the button?

File: src/elements/Button/Button.module.sass [hunk 9]
```diff
@@ -0,0 +1,238 @@
+//!variant
```
Comment (MaksPapirovnyk) [inline src/elements/Button/Button.module.sass:1]: looks very messy, with a lot of `important` fragile styles. 
this needs refactoring, add TODO about this.

**Review by MaksPapirovnyk — CHANGES_REQUESTED** [review]
IN THE DESCRIPRICTION OF THE TICKET USE SCEREENSHOTS OF FINAL IMPLEMENTATION OF THE UI, FOR EXAMPLE SHOW THE VARIANTS OF CREATED BUTTONS! NOT CODE! NOT CODE! NOT CODE! THE CODE IN THE PR :)

**Comment by MaksPapirovnyk** [comment]
We don't need button for all occasions, we need button with clear understanding wich variants we have

**Comment by MaksPapirovnyk** [comment]
Please resolve conflicts

File: src/elements/Button/Button.tsx (hunk 7 above)
Comment (dipSyxx) [reply src/elements/Button/Button.tsx:9]: it mean link with text-decoration: underline :)

Comment (dipSyxx) [reply src/elements/Button/Button.tsx:9]: i already fixed it :)

File: src/elements/Button/Button.tsx (hunk 1 above)
Comment (dipSyxx) [reply src/elements/Button/Button.tsx:8]: fixed, now we can insert only element a (if we have link) and button

**Review by MaksPapirovnyk — APPROVED** [review]
LGTM

**Pull request merged**